logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/session/RegisterTokenUseCase.kt

```diff
@@ -1,32 +1,45 @@
 package com.wire.kalium.logic.feature.session
 
 import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.configuration.notification.NotificationTokenRepository
 import com.wire.kalium.logic.data.client.ClientRepository
 import com.wire.kalium.logic.functional.fold
 import com.wire.kalium.network.api.user.pushToken.PushTokenBody
 import com.wire.kalium.network.exceptions.KaliumException
 import com.wire.kalium.network.exceptions.isInvalidCode
+import com.wire.kalium.network.exceptions.isNotFound
 
-class RegisterTokenUseCase(
-    private val clientRemoteRepository: ClientRepository
-) {
-    suspend operator fun invoke(senderId: String, clientId: String, token: String, transport: String): RegisterTokenResult =
-        clientRemoteRepository.registerToken(
-            body = PushTokenBody(
-                senderId = senderId, client = clientId, token = token, transport = transport
-            )
-        ).fold({
-            if (
-                it is NetworkFailure.ServerMiscommunication &&
-                it.kaliumException is KaliumException.InvalidRequestError &&
-                it.kaliumException.isInvalidCode()
-            ) {
-                RegisterTokenResult.Failure.AppNotFound
-            } else {
-                RegisterTokenResult.Failure.Generic(it)
-            }
-        }, {
-            RegisterTokenResult.Success
+interface RegisterTokenUseCase {
+    suspend operator fun invoke(senderId: String, clientId: String): RegisterTokenResult
+}
+
+class RegisterTokenUseCaseImpl(
+    private val clientRemoteRepository: ClientRepository,
+    private val notificationTokenRepository: NotificationTokenRepository
+) : RegisterTokenUseCase {
+    override suspend operator fun invoke(senderId: String, clientId: String): RegisterTokenResult =
+        notificationTokenRepository.getNotificationToken().fold({
+            RegisterTokenResult.Failure.NotificationTokenNotFound(it)
+        }, { notificationToken ->
+            clientRemoteRepository.registerToken(
+                body = PushTokenBody(
+                    senderId = senderId, client = clientId, token = notificationToken.token, transport = notificationToken.transport
+                )
+            ).fold({ failure ->
+                RegisterTokenResult.Success
+                if (failure is NetworkFailure.ServerMiscommunication && failure.kaliumException is KaliumException.InvalidRequestError)
+                    when {
+                        failure.kaliumException.isInvalidCode() -> RegisterTokenResult.Failure.AppNotFound
+                        failure.kaliumException.isNotFound() -> RegisterTokenResult.Failure.PushTokenRegister
```

Comment:
@MohammedMokresh knows about this one more than me, but i think AppNotFound is on 404 response